server/src/main/java/org/apache/gravitino/server/web/rest/TableOperations.java

```diff
@@ -84,6 +89,16 @@ public Response listTables(
           () -> {
             Namespace tableNS = NamespaceUtil.ofTable(metalake, catalog, schema);
             NameIdentifier[] idents = dispatcher.listTables(tableNS);
+            idents =
+                MetadataFilterHelper.filterByExpression(
+                    metalake,
+                    "((ANY(USE_CATALOG,METALAKE,CATALOG,SCHEMA)) && "
+                        + "(ANY(USE_SCHEMA,METALAKE,CATALOG,SCHEMA)) &&"
+                        + " ( ANY(SELECT_TABLE,METALAKE,CATALOG,SCHEMA,TABLE) || "
+                        + " ANY(MODIFY_TABLE,METALAKE,CATALOG,SCHEMA,TABLE|| TABLE::OWNER)) ||"
```

Comment:
Sorry for my mistake. Table:Owner shouldn't be included in `ANY`.

Reply:
fixed